qtcodes/fitters/repetition.py

```diff
@@ -20,88 +17,14 @@ class RepetitionDecoder(LatticeDecoder):
-                left = (t, j - 0.5, 0)
-                right = (t, j + 0.5, 0)
-                self.S["Z"].add_edge(
-                    self.node_map["Z"][left], self.node_map["Z"][right], edge_weight
-                )
-
-            # edges (real-virtual)
-            self.S["Z"].add_edge(
-                self.node_map["Z"][self.virtual["Z"][0]],
-                self.node_map["Z"][(t, 0.5, 0)],
-                edge_weight,
-            )  # left
-            self.S["Z"].add_edge(
-                self.node_map["Z"][self.virtual["Z"][1]],
-                self.node_map["Z"][(t, self.params["d"] - 1.5, 0)],
-                edge_weight,
-            )  # right
-
-        # connect physical qubits in same location across subgraphs of adjacent times
-        syndrome_nodes_t0 = [(t, x, y) for t, x, y in self.S["Z"].nodes() if t == 0]
-        for node in syndrome_nodes_t0:
-            space_label = (node[1], node[2])
-            for t in range(0, self.params["T"] - 1):
-                self.S["Z"].add_edge(
-                    self.node_map["Z"][(t,) + space_label],
-                    self.node_map["Z"][(t + 1,) + space_label],
-                    edge_weight,
-                )
-
-    def _specify_virtual(self) -> Dict[str, List[TQubit]]:
-        """
-        Define coordinates of P virtual syndrome nodes
-        (i.e. parity measurements to which we don't have access),
-
-        Args:
-        Returns:
-            virtual (dictionary): where virtual["Z"] holds a list of tuples
-            specifying virtual P syndrome nodes
-        """
-        virtual: Dict[str, List[TQubit]] = {}
-        virtual["Z"] = []
-        virtual["Z"].append((-1, -0.5, 0))
-        virtual["Z"].append((-1, self.params["d"] - 0.5, 0))
-        return virtual
-
-    def _is_crossing_readout_path(
-        self, match: Tuple[TQubit, TQubit], logical_readout_type: str
-    ) -> bool:
-        """
-        Helper method that detects whether the match is crossing the readout path.
-
-        Args:
-            match (Tuple[TQubit, TQubit]): match in MWPM between two nodes
-            logical_readout_type (str): e.g. "X"
-
-        Returns:
-            (bool): whether or not the match is crosing the readout path
-        """
-        source, target = match
-        if logical_readout_type == "Z":
-            return (source[0] == -1 and source[1] == -0.5) or (
-                target[0] == -1 and target[1] == -0.5
-            )  # top
+    def _params_validation(self):
+        # The repetition code is a 1D lattice
+        if "d" not in self.params:
+            self.params["d"] = (3, 1)
+        elif isinstance(self.params["d"], Number):
+            d = int(self.params["d"])
+            self.params["d"] = (d, 1)
         else:
-            raise NotImplementedError(
-                "Only Z readout is supported in the Repetition code."
-            )
+            self.params["d"] = (self.params["d"][0], 1)
```

Comment:
This allows the user to pass in `"d": (3,2)` and still end up with `self.params["d"] = (3,1)`. Let's make sure that `self.params["d"][1] == 1` and otherwise raise an exception. Also, let's use the constants you made, instead of `[0]` and `[1]`. 

We were briefly chatting about a (phase-flip-protected) repetition qubit protected against Z flips. On a related note, we could allow users to pass in `"d": (1,3), "phase-flip-protected": True` to `RepetitionQubit` when they want to create a  phase-flip-protected repetition qubit. And, in the case where a user just sends in `"d": (1,3)` without setting `"phase-flip-protected": True`, we can raise a warning stating that the user is creating a phase-flip-protected repetition qubit instead of a possibly desired bit-flip-protected repetition qubit. 

To enable this phase-flip-protected functionality (tracked by issue #66), I believe we will just need to modify this parameter validation and add some tests. However, we can leave that for a separate PR, but feel free to implement the phase-flip-protected functionality in this PR.

Reply:
This is fixed now